packages/yoastseo/src/languageProcessing/helpers/word/getAllWordsFromTree.js

```diff
@@ -0,0 +1,23 @@
+import getSentencesFromTree from "../sentence/getSentencesFromTree";
+import { flatMap } from "lodash-es";
+import removePunctuation from "../sanitize/removePunctuation";
+
+/**
+ * Gets the words from the tree, i.e. the paragraph and heading nodes.
+ * These two node types are the nodes that should contain words for the analysis.
+ *
+ * @param {Paper} paper The paper to get the tree and words from.
+ *
+ * @returns {String[]} Array of words retrieved from the tree.
+ */
+export default function( paper ) {
+	const sentences = getSentencesFromTree( paper );
+	// Get all the tokens from each sentence.
+	const tokens = sentences.map( sentence => sentence.tokens );
+	let words = flatMap( tokens ).map( token => token.text );
+	// Remove punctuation and spaces.
+	words = words.map( token => removePunctuation( token ) );
```

Comment:
That's indeed an interesting idea, though definitely should be tackled in a different issue. Here we need to filter out the punctuation and spaces because it doesn't make sense to include them in the word length calculation. Your suggestion could help make this process neater indeed

Reply:
Yep, great suggestion, let's not do that in this PR though 😅 Could one of you create a new task?

Reply:
I'll create the task :)